tiledb/sm/array_schema/array_schema.cc

```diff
@@ -84,8 +84,11 @@ ArraySchema::ArraySchema(ArrayType array_type)
 ArraySchema::ArraySchema(const ArraySchema* array_schema) {
   array_uri_ = array_schema->array_uri_;
   array_type_ = array_schema->array_type_;
-  attributes_ = array_schema->attributes_;
-  attribute_num_ = array_schema->attribute_num();
+  // Copy attributes using add_attribute
+  for (auto attribute : array_schema->attributes_)
+    add_attribute(attribute);
+  // Set domain
+  set_domain(array_schema->domain_);
```

Comment:
`set_domain()` calls `delete domain_;` and at this point I think `domain_` is uninitialized. Do we need to set it to `nullptr` before calling `set_domain()`?

Reply:
Good catch, yes domain_ needs to be set to nullptr before calling set_domain(). I have corrected this.